scripts/feature_tests/upgrade/clean_setup_env.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+METAL3_DEV_ENV_DIR="$(dirname "$(readlink -f "${0}")")/../../../"
+
+# Deploy a fresh metal3-dev-env after each test case
+# to overcome environmental flakiness
+pushd "${METAL3_DEV_ENV_DIR}" || exit
+make clean
```

Comment:
Maybe you could put this in a single loop?

Reply:
True. This was just a workaround to re-run 'make' once in case of failure.

I think it should be discussed separately how to handle 'make' problems since those impact the whole CI. The logic should be put to some common directory then.

Those are the reasons why I would not change this workaround.

Reply:
> True. This was just a workaround to re-run 'make' once in case of failure.
> 
> I think it should be discussed separately how to handle 'make' problems since those impact the whole CI. The logic should be put to some common directory then.

But whole CI, do you mean upgrade CI jobs?
> Those are the reasons why I would not change this workaround.

Reply:
No, I mean the airship CI. It's rather often that a CI job needs re-triggering manually due to an environment deployment failure.